include/filewriter/summarywriter_impl.hpp

```diff
@@ -150,6 +150,97 @@ void SummaryWriter<Filewriter>::Image(const std::string& tag,
     remove(fileNames[i].c_str());
 }
 
+template<typename Filewriter>
+void SummaryWriter<Filewriter>::Histogram(const std::string& tag,
+                                          int step,
+                                          std::vector<double>& values,
+                                          std::vector<double>& bins,
+                                          Filewriter& fw)
+{
+  size_t num = values.size();
+  double min = std::numeric_limits<double>::max();
+  double max = std::numeric_limits<double>::lowest();
+  double sum = 0.0;
+  double sumofSquares = 0.0;
+
+  std::vector<int> counts(bins.size(), 0);
+  for (size_t i = 0; i < num; ++i)
+  {
+    float v = values[i];
+    auto lb =
+      lower_bound(bins.begin(), bins.end(), v);
```

Comment:
```suggestion
    auto lb = lower_bound(bins.begin(), bins.end(), v);
```

It think it might be getting covered in single line.
If not then
```suggestion
    auto lb = lower_bound(bins.begin(), bins.end(),
        v);
```

Is a better way.